compiler/rustc_middle/src/mir/traversal.rs

```diff
@@ -307,3 +306,91 @@ pub fn reachable_as_bitset<'tcx>(body: &Body<'tcx>) -> BitSet<BasicBlock> {
     (&mut iter).for_each(drop);
     iter.visited
 }
+
+#[derive(Clone)]
+pub struct ReversePostorderCache<'a, 'tcx> {
```

Comment:
Maybe `ReversePostorderIter` or to keep existing naming convention `CachedReversePostorder`? The struct is not a cache by itself.